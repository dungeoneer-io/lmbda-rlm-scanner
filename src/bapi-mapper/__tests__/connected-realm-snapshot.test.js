const { describe, expect, test } = require('@jest/globals');

const {
    convertIndexResultsToCrlmSnapshot,
    extractAllRealmsFromCrlmSnapshot
} = require('../connected-realm-snapshot');

const ArrayOfConnectedRealms = require('./__fixtures__/array-of-connected-realms.json');
const ConnectedRealmSnapshot = require('./__fixtures__/connected-realm-snapshot.json');

describe('convertIndexResultsToCrlmSnapshot', () => {
    test('converts array to nested object in snapshot', () => {
        jest.useFakeTimers()
            .setSystemTime(new Date('1983-10-08 14:10'));
        const actual = convertIndexResultsToCrlmSnapshot(ArrayOfConnectedRealms);
        expect(actual).toMatchSnapshot();
    });
});

describe('extractAllRealmsFromCrlmSnapshot', () => {
    test('flatmaps out the realms in the snapshot', () => {
        const result = extractAllRealmsFromCrlmSnapshot(ConnectedRealmSnapshot);
    
        expect(result).toMatchSnapshot();
    });
});
