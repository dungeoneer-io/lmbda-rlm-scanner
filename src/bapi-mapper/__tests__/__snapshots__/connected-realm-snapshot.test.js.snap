// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertIndexResultsToCrlmSnapshot converts array to nested object in snapshot 1`] = `
{
  "data": [
    {
      "id": 11,
      "realms": [
        {
          "category": "United States",
          "crlm": 11,
          "id": 11,
          "locale": "enUS",
          "name": "Tichondrius",
          "region": 1,
          "slug": "tichondrius",
          "timezone": "America/Los_Angeles",
          "type": "NORMAL",
        },
      ],
    },
    {
      "id": 99,
      "realms": [
        {
          "category": "United States",
          "crlm": 99,
          "id": 99,
          "locale": "enUS",
          "name": "Llane",
          "region": 1,
          "slug": "llane",
          "timezone": "America/New_York",
          "type": "NORMAL",
        },
        {
          "category": "United States",
          "crlm": 99,
          "id": 1297,
          "locale": "enUS",
          "name": "Arygos",
          "region": 1,
          "slug": "arygos",
          "timezone": "America/New_York",
          "type": "NORMAL",
        },
      ],
    },
    {
      "id": 1070,
      "realms": [
        {
          "category": "United States",
          "crlm": 1070,
          "id": 1070,
          "locale": "enUS",
          "name": "Alexstrasza",
          "region": 1,
          "slug": "alexstrasza",
          "timezone": "America/Chicago",
          "type": "NORMAL",
        },
        {
          "category": "United States",
          "crlm": 1070,
          "id": 1563,
          "locale": "enUS",
          "name": "Terokkar",
          "region": 1,
          "slug": "terokkar",
          "timezone": "America/Chicago",
          "type": "NORMAL",
        },
      ],
    },
  ],
  "stamp": 434484600000,
  "type": "ConnectedRealmList",
}
`;

exports[`extractAllRealmsFromCrlmSnapshot flatmaps out the realms in the snapshot 1`] = `
[
  {
    "category": "United States",
    "crlm": 11,
    "id": 11,
    "locale": "enUS",
    "name": "Tichondrius",
    "region": 1,
    "slug": "tichondrius",
    "timezone": "America/Los_Angeles",
    "type": "NORMAL",
  },
  {
    "category": "United States",
    "crlm": 99,
    "id": 99,
    "locale": "enUS",
    "name": "Llane",
    "region": 1,
    "slug": "llane",
    "timezone": "America/New_York",
    "type": "NORMAL",
  },
  {
    "category": "United States",
    "crlm": 99,
    "id": 1297,
    "locale": "enUS",
    "name": "Arygos",
    "region": 1,
    "slug": "arygos",
    "timezone": "America/New_York",
    "type": "NORMAL",
  },
  {
    "category": "United States",
    "crlm": 1070,
    "id": 1070,
    "locale": "enUS",
    "name": "Alexstrasza",
    "region": 1,
    "slug": "alexstrasza",
    "timezone": "America/Chicago",
    "type": "NORMAL",
  },
  {
    "category": "United States",
    "crlm": 1070,
    "id": 1563,
    "locale": "enUS",
    "name": "Terokkar",
    "region": 1,
    "slug": "terokkar",
    "timezone": "America/Chicago",
    "type": "NORMAL",
  },
]
`;
